src/transformers/models/qdqbert/modeling_qdqbert.py

```diff
@@ -0,0 +1,1744 @@
+                attention_scores = attention_scores + relative_position_scores
+            elif self.position_embedding_type == "relative_key_query":
+                relative_position_scores_query = torch.einsum("bhld,lrd->bhlr", query_layer, positional_embedding)
+                relative_position_scores_key = torch.einsum("bhrd,lrd->bhlr", key_layer, positional_embedding)
+                attention_scores = attention_scores + relative_position_scores_query + relative_position_scores_key
+
+        attention_scores = attention_scores / math.sqrt(self.attention_head_size)
+        if attention_mask is not None:
+            # Apply the attention mask is (precomputed for all layers in QDQBertModel forward() function)
+            attention_scores = attention_scores + attention_mask
+
+        # Normalize the attention scores to probabilities.
+        attention_probs = nn.Softmax(dim=-1)(attention_scores)
+
+        # This is actually dropping out entire tokens to attend to, which might
+        # seem a bit unusual, but is taken from the original Transformer paper.
+        attention_probs = self.dropout(attention_probs)
+
+        # Mask heads if we want to
+        if head_mask is not None:
+            attention_probs = attention_probs * head_mask
+
+        context_layer = torch.matmul(
+            self.matmul_a_input_quantizer(attention_probs), self.matmul_v_input_quantizer(value_layer)
+        )
+
+        context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
+        context_layer = context_layer.view(*new_context_layer_shape)
+
+        outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+
+        if self.is_decoder:
+            outputs = outputs + (past_key_value,)
+        return outputs
+
+
+class QDQBertSelfOutput(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        # Quantize Linear layer
+        self.dense = quant_nn.QuantLinear(config.hidden_size, config.hidden_size)
+
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+        # Quantize the inputs to the residual add
+        self.add_local_input_quantizer = TensorQuantizer(quant_nn.QuantLinear.default_quant_desc_input)
+        self.add_residual_input_quantizer = TensorQuantizer(quant_nn.QuantLinear.default_quant_desc_input)
+
+    def forward(self, hidden_states, input_tensor):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        # Quantize the inputs to the residual add
+        add_local = self.add_local_input_quantizer(hidden_states)
+        add_residual = self.add_residual_input_quantizer(input_tensor)
+        hidden_states = self.LayerNorm(add_local + add_residual)
+        return hidden_states
+
+
+class QDQBertAttention(nn.Module):
```

Comment:
This looks identical to the BertAttention layer with just a renaming, so could you add a
```
# Copied from transformers.modeling.bert.modeling_bert.BertAttention with Bert -> QDQBert
```
before the class definition? That way the copy will stay consistent with the original.

This also applies to any other class that is a direct copy of Bert, I will try to flag them.

Reply:
addressed in https://github.com/huggingface/transformers/pull/14066/commits/4ffbf92ab21b3ce9e646a3b0d60d22e8063765d3

I have some questions related to the "copied from" statement:
1. Besides the classes mentioned in your comments, I also added the "copied from" statement to QDQBertEmbeddings and QDQBertPreTrainedModel. Does it make sense?

2. Some of classes which are added with the "copied from" statement are not exactly copied from BERT. (with minor changes) Those minor changes are for example changing from 
`self.attention = BertAttention(config)` to `self.attention = QDQBertAttention(config)`
in class QDQBertLayer. Does that matter? Can we still add the "copied from" statement to those classes?

Reply:
I managed to run `make fix-copies` and I think I understand this better now.

For question#1, the "copied from" statement will be only added to transformers modules. In this case, QDQBertEmbeddings will be added with the "copied from" statement.

For question#2, I will only add "copied from" statement for functions that are exactly copied from BERT. Those functions with minor changes, the "copied from" statement should not be applied. Instead, I used "Based on" ahead of those classes.

A newer commit is addressed: https://github.com/huggingface/transformers/pull/14066/commits/6dd6567353ed335cf32a8ba8e79b864686827b11

Reply:
That sounds good, thank you @shangz-ai!